pkg/kubectl/resource/builder.go

```diff
@@ -567,25 +567,34 @@ func (b *Builder) visitorResult() *Result {
 }
 
 func (b *Builder) visitBySelector() *Result {
+	result := &Result{
+		targetsSingleItems: false,
+	}
+
 	if len(b.names) != 0 {
-		return &Result{err: fmt.Errorf("name cannot be provided when a selector is specified")}
+		result.err = fmt.Errorf("name cannot be provided when a selector is specified")
```

Comment:
These methods are expected to be single path invocations with err dominating the return (an error short circuits).  Callers are expected to handle error first, rather than later.  So not altering the target state with an error is just another form of short-circuit (which is how single item implied worked today).